python/feldera/pipeline.py

```diff
@@ -692,9 +704,50 @@ def query(self, query: str) -> Generator[Mapping[str, Any], None, None]:
         :raises FelderaAPIError: If querying a non materialized table or view.
         :raises FelderaAPIError: If the query is invalid.
         """
+        warnings.warn(
+            "function query is deprecated and will be removed in a future version",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
 
         return self.client.query_as_json(self.name, query)
 
+    def query_pyarrow(self, query: str) -> pyarrow.Table:
+        """
+        Executes an ad-hoc SQL query on this pipeline and returns a
+        :class:`.pyarrow.Table` containing all the results.
+
+        Note:
+            You can only ``SELECT`` from materialized tables and views.
+
+        :param query: The SQL query to be executed.
+        :param path: The path of the parquet file.
+
+        :raises FelderaAPIError: If the pipeline is not in a RUNNING or PAUSED state.
+        :raises FelderaAPIError: If querying a non materialized table or view.
+        :raises FelderaAPIError: If the query is invalid.
+        """
+
+        return self.client.query_as_pyarrow(self.name, query)
+
+    def query_pylist(self, query: str) -> List[Mapping[str, Any]]:
```

Comment:
it seems like given this return type suffers from the same problem as the json (aliased columns will be removed?)

Reply:
Ok probably not a good return type then, can we remove it?

Its stupid but unless we change our SQL to not accept this nonsense I dont see a good alternative cc @mihaibudiu

Reply:
maybe instead if str as key you can have a Column Type that hashes using unique identifiers

Reply:
no need to remove but either use a key that has a unique hash like the position or list of list